### ISSUE TITLE
more helpful error messages

### ISSUE THREAD

**Issue by ronan-mch** [projectblacklight/blacklight#1377]
When an invalid request is sent to Solr, [an exception and stack trace is sent to the user](https://github.com/projectblacklight/blacklight/blob/v5.16.4/lib/blacklight/solr/repository.rb#L55). This is often extremely long and hard to parse. Solr does however provide useful error messages in the `msg` key of the response, somewhere towards the bottom of the exception. It would be useful to have that message printed first in the exception if it exists.

I guess the correct way to fix this would be by adding a method to RSolr which Blacklight could then call, if someone has a suggestion I'm happy to create a PR.

**Comment by jcoyne** [comment]
@ronan-mch Yeah, that would be much better. Please send a PR.

**Comment by ronan-mch** [comment]
Do you agree it should be in rsolr or is it easier to parse it in
blacklight? My concern is that the rsolr error class won't be aware of what
type of response it is and thus won't be able to parse it in a simple way.

ons. 27. jan. 2016 17.37 skrev Justin Coyne <notifications@github.com>:

**Comment by cbeer** [comment]
RSolr makes sense to me; it looks like there's even some primitive attempt at parsing the body already: https://github.com/rsolr/rsolr/blob/master/lib/rsolr/error.rb#L25

**Comment by atz** [comment]
Resolved in rsolr, afaict.
